botocore/client.py

```diff
@@ -373,6 +379,16 @@ def _check_default_region(self, service_name, region_name):
         if self.client_config and self.client_config.region_name is not None:
             return self.client_config.region_name
 
+    def _is_regionalized_endpoint(self, service_name, partition_name='aws'):
```

Comment:
I'd prefer to not be accessing the `_endpoint_data`. Looking through the code, I'm not opposed to exposing a general `is_regionalized` method on the `EndpointResovler` class. Specifically:

1. The endpoint resolver is marked as an internal component on the session and that is the only location a user can hook into the resolver.
2. The purpose of the resolver is to be a utility for interpreting the endpoints file. This logic is really only interpreting the endpoints file. Plus it is actually useful functionality to have (i.e. I can see it being exposed on the session in the future like we do for `get_available_regions`).